libs/blocks/uar-results/uar-results.js

```diff
@@ -1,59 +1,94 @@
     el.append(a);
     const { default: createFragment } = await import('../fragment/fragment.js');
     await createFragment(a);
   }
 }
 
-function redirectPage(quizUrl) {
+function redirectPage(quizUrl, debug, message) {
   const url = (quizUrl) ? quizUrl.text : 'https://adobe.com';
-  console.log(`Invalid Result, redirecting to: ${url}`);
+  window.lana.log(message);
+
+  if (debug === 'uar') {
+    // eslint-disable-next-line no-console
+    console.log(`${message}, redirecting to: ${url}`);
+  } else {
+    window.location = url;
+  }
+}
+
+function setAnalytics(hashValue, debug) {
+  /* eslint-disable no-underscore-dangle */
+  window.alloy_load ??= {};
+  window.alloy_load.data ??= {};
+  window.alloy_all ??= {};
+  window.alloy_all.data ??= {};
+  window.alloy_all.data._adobe_corpnew ??= {};
+  window.alloy_all.data._adobe_corpnew.digitalData ??= {};
+  window.alloy_all.data._adobe_corpnew.digitalData.page ??= {};
+  window.alloy_all.data._adobe_corpnew.digitalData.page.pageInfo ??= {};
+  window.alloy_all.data._adobe_corpnew.digitalData.page.pageInfo.customHash = hashValue;
+  if (debug === 'uar') {
+    // eslint-disable-next-line no-console
+    console.log(`Setting a custom hash for pageload to: ${window.alloy_all.data._adobe_corpnew.digitalData.page.pageInfo.customHash}`);
+  }
 }
 
-export default async function init(el) {
+export default async function init(el, debug = null, localStoreKey = null) {
   const data = getMetadata(el);
   const metaData = el.querySelectorAll('div');
   const params = new URL(document.location).searchParams;
-  const localStoreKey = params.get('quizKey');
   const quizUrl = data['quiz-url'];
   const BASIC_KEY = 'basicFragments';
   const NESTED_KEY = 'nestedFragments';
+  const HASH_KEY = 'pageloadHash';
 
-  let results = localStorage.getItem(localStoreKey);
-
-  for( const div of metaData){
-    div.remove();
-  }
+  /* eslint-disable no-param-reassign */
+  // handle these two query param values in this way to facilitate unit tests
+  localStoreKey ??= params.get('quizKey');
+  debug ??= params.get('debug');
 
+  let results = localStorage.getItem(localStoreKey);
   if (!results) {
-    redirectPage(quizUrl);
-    return
+    redirectPage(quizUrl, debug, 'Could not load UAR results: local storage missing');
+    return;
   }
-
   results = JSON.parse(results);
 
-  if(data['nested-fragments'] && el.classList.contains('nested')) {
+  if (data['nested-fragments'] && el.classList.contains('nested')) {
     const nested = results[NESTED_KEY][data['nested-fragments'].text];
     if (nested) loadFragments(el, nested);
   } else if (el.classList.contains('basic')) {
-    const structure = results[BASIC_KEY];
-    if (!structure) { 
-      redirectPage(quizUrl);
-      return
+    const basic = results[BASIC_KEY];
+    const pageloadHash = results[HASH_KEY];
+
+    if (!basic) {
+      redirectPage(quizUrl, debug, 'Could not load UAR results: Basic fragments are missing');
```

Comment:
if you save some of these strings ("Could not load...") as variables, you could export them and then import them into your tests to ensure that a change to the string in the file does not break the test.